guides/source/active_record_encryption.md

```diff
@@ -0,0 +1,420 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**
+
+Active Record Encryption
+========================
+
+This guide covers encrypting your database information using Active Record.
+
+After reading this guide you will know:
+
+* How to set up database encryption with Active Record.
+* How to migrate unencrypted data
+* How to make different encryption schemes coexist
+* How to use the API
+* How to configure the library and how to extend it
+
+--------------------------------------------------------------------------------
+
+Active Record supports application-level encryption. It works by declaring which attributes should be encrypted and seamlessly encrypting and decrypting them when necessary. The encryption layer is placed between the database and the application. The application will access unencrypted data but the database will store it encrypted.
+
+## Basic usage
+
+### Setup
+
+First, you need to add some keys to your [rails credentials](/security.html#custom-credentials). Run `bin/rails db:encryption:init` to generate a random key set:
+
+```bash
+$ bin/rails db:encryption:init
+Add this entry to the credentials of the target environment:
+
+active_record.encryption:
+  master_key: EGY8WhulUOXixybod7ZWwMIL68R9o5kC
+  deterministic_key: aPA5XyALhf75NNnMzaspW7akTfZp0lPY
+  key_derivation_salt: xEY0dt6TZcAMg52K7O84wYzkjvbA62Hz
+```
+
+NOTE: These generated keys and salt are 32 bytes length. If you generate these yourself, the minimum lengths you should use are 12 bytes for the master key (this will be used to derive the AES 32 bytes key) and 20 bytes for the salt.
+
+### Declaration of encrypted attributes
+
+Encryptable attributes are defined at the model level. These are regular Active Record attributes backed by a column with the same name. 
+
+```ruby
+class Article < ApplicationRecord
+  encrypts :title
+end
+````
+
+The library will transparently encrypt these attributes before saving them into the database, and will decrypt them when retrieving their values:
+
+```ruby
+article = Article.create title: "Encrypt it all!"
+article.title # => "Encrypt it all"
```

Comment:
I assume the title output is supposed to be the same as the input?

```suggestion
article.title # => "Encrypt it all!"
```

Reply:
Indeed 😅. Ty!